### DISCUSSION TITLE
Support for Glossary entries

### DISCUSSION THREAD

**Discussion by LecrisUT** [retorquere/zotero-better-bibtex#1853] [Ideas]
Given that this supports bib export, could it also support generating bib files for `bib2gls`?

The relevant bib format for these entries is found in Section 4 of the [documentation](http://mirrors.ctan.org/support/bib2gls/bib2gls.pdf). Zotero does not have a specific item for glossaries, but can custom item types be created for this?

**Comment by retorquere** [comment]
> Given that this supports bib export, could it also support generating bib files for `bib2gls`?
> 
> The relevant bib format for these entries is found in Section 4 of the [documentation](http://mirrors.ctan.org/support/bib2gls/bib2gls.pdf).

I've given that a quick scan, but I'm not sure what's being asked here. What would an export of these kinds of entries use as a source?

> Zotero does not have a specific item for glossaries, but can custom item types be created for this?

New Zotero item types? That's not something plugins can do.

**Reply by LecrisUT** [reply]
There was an attempt at hacking these entries at  chrisjsewell/bib2glossary. There a hack of some entries to map `label`, `name`, etc. on the bib file with `title`, `extra` etc. on zotero to create a compatible bib file.

The usecase for this would be to easily share a convention and standard notations between LaTeX projects and collaborators.

For reference here's how it works in LaTeX.
Some simple glossary entries you would have are like:
```
@entry{Label1,    %Similar to cite keys
  name = {name},
  description = {long description}
}
@abbreviation{Label2,
  short = {n},
  long = {name}
}
```
In latex these will create a table of glossaries like:
```
   n     .......... name
   name  .......... long description
```

**Reply by retorquere** [reply]
That's OK, but what would the source for this be in Zotero?

**Reply by LecrisUT** [reply]
Ideally having custom entries would be preferred, but as that's not possible yet, maybe use a similar strategy to chrisjsewell/bib2glossary:

So these would be a list of `Dictionary Entries` with a mapping like:
| Zotero     | Bib              |
|------------|------------------|
| Title      | name/short       |
| Abstract   | description/long |
| Extra:Type | type             |

**Reply by retorquere** [reply]
I'm still not sure where/how you'd store this in Zotero. This is starting to sound like an enhancement request -- can you open an issue for this?

**Reply by LecrisUT** [reply]
On the `better-bibtex` side, these would be stored in a separate Collection so they can be exported to a different `.bib` file for LaTeX. I'll make the issue to keep track of this.
